pkg/controller/storagecluster/update.go

```diff
@@ -740,9 +741,9 @@ func isCSIBounceRequired(oldSpec, currentSpec *corev1.StorageClusterSpec) bool {
 		return oldSpec.CSI.Enabled != currentSpec.CSI.Enabled
 	}
 
-	// New CSI spec added to cluster. If enabled, pod will bounce
-	if oldSpec.CSI == nil && currentSpec.CSI != nil {
-		return currentSpec.CSI.Enabled
+	// Spec changed from enabled -> disabled or vice versa
+	if isCSIEnabled(oldSpec) != isCSIEnabled(currentSpec) {
```

Comment:
nit: can we assume for currentSpec we just need to check the new field?
```
if isCSIEnabled(oldSpec) != currentSpec.CSI
```

Reply:
We could, but there's always the case where a user re-adds the feature gate even though it's deprecated. That'll be reconciled to the CSI.Enabled field. Just wanted to be safe and cover our bases here.